src/main/java/fi/digitraffic/tis/vaco/ui/UiController.java

```diff
@@ -267,12 +272,22 @@ public ResponseEntity<Resource<ImmutableEntryState>> fetchEntryState(@PathVariab
                 });
                 List<Summary> summaries = entryStateService.getTaskSummaries(entry);
                 Optional<Company> company = companyHierarchyService.findByBusinessId(entry.businessId());
+
+                String credentialPublicId = entry.credentials();
+                Credentials entrysCredential = null;
+
+                if (credentialPublicId != null) {
+                    Optional<Credentials> credentials = credentialsService.findByPublicId(credentialPublicId);
+                    entrysCredential= credentials.get();
+                }
```

Comment:
All this could be compressed into
```java
.credentials(credentialsService.findbyPublicId(entry.credentials().orElse(null))
```

This does require modifying `CredentialsService#findByPublicId` just a bit like this
```java
    public Optional<Credentials> findByPublicId(String publicId) {
        return Optional.ofNullable(publicId)
            .map(credentialsRepository::findByPublicId)
            .map(this::asCredentialsEntity);
    }
```

but that wouldn't be a terrible adjustment either.